conans/client/cmd/test.py

```diff
@@ -36,7 +36,8 @@ def install_build_and_test(app, conanfile_abs_path, reference, graph_info,
                      manifest_interactive=manifest_interactive,
                      keep_build=keep_build,
                      recorder=recorder)
-        cmd_build(app, conanfile_abs_path, base_folder, test_build_folder, package_folder=None,
+        cmd_build(app, conanfile_abs_path, base_folder, test_build_folder,
+                  package_folder=os.path.join(test_build_folder, "package"),
```

Comment:
I would use from conans.paths.PACKAGES_FOLDER instead:

```python
from conans.paths import PACKAGES_FOLDER

...

cmd_build(app, conanfile_abs_path, base_folder, test_build_folder,
                  package_folder=os.path.join(test_build_folder, PACKAGES_FOLDER),
```

Reply:
I think it is better to leave it explicit and independent, that is an internal path definition for the Conan cache. We might want to change it anytime, to something like "pkg", but we don't want to break what happens in user space.